feat: provide `faucet upgrade` command

jacderida opened the pull request:
Uses the generic service manager to provide a `faucet upgrade` command, which will be used during the general upgrade process to either upgrade or downgrade the faucet.

Some code was common between the node and faucet upgrades, so that was refactored to be shared between the two.

Also clarifies some of the language used on the command documentation.

## Description

reviewpad:summary

maqi commented on sn_dysfunction/src/detection.rs:
```diff
@@ -73,100 +81,111 @@ impl DysfunctionDetection {
 
-        let nodes = self.nodes.to_vec();
+        let nodes = &self.nodes;
         for node in nodes.iter() {
```
you can then use `for node in &self.nodes` directly here?

maqi commented on sn_dysfunction/src/detection.rs:
```diff
@@ -73,100 +81,111 @@ impl DysfunctionDetection {
+            return 0.0;
+        }
+
+        debug!("node {node} issue count: {:?}", node_issue_count);
```
for easier tracking, better log the type of the issue together here?

maqi commented on sn_dysfunction/src/detection.rs:
```diff
@@ -962,7 +1006,7 @@ mod knowledge_tests {
         // Add a new nodes
         let new_node = random_xorname();
-        dysfunctional_detection.add_new_node(new_node);
+        // dysfunctional_detection.add_new_node(new_node);
```
no longer to call the separate `add_new_node` ? 
if so, just remove the line?

maqi commented on sn_dysfunction/src/lib.rs:
```diff
@@ -170,6 +177,24 @@ impl DysfunctionDetection {
             }
         }
     }
+    /// Removes a probe tracker from the node liveness records. Returns true if a record was removed
```
no boolean returned.
comment need to be changed.

maqi commented on sn_dysfunction/src/detection.rs:
```diff
@@ -216,23 +229,16 @@ impl DysfunctionDetection {
-                        DIFF_THRESHOLD
-                    };
+            let zscore = {
+                let threshold = std_dev.unwrap_or(1.0);
```
the std_dev is measuring the `diff to the average` 
so I think the `threshold` is actually shall be something like `average/mean + std_dev` or `average/mean + std_dev/2` ?

jacderida committed:
feat: provide `faucet upgrade` command

Uses the generic service manager to provide a `faucet upgrade` command, which will be used during
the general upgrade process to either upgrade or downgrade the faucet.

Some code was common between the node and faucet upgrades, so that was refactored to be shared
between the two.

Also clarifies some of the language used on the command documentation.
